cirq-core/cirq/circuits/circuit_operation.py

```diff
@@ -674,4 +680,5 @@ def with_params(
     def _resolve_parameters_(
         self, resolver: 'cirq.ParamResolver', recursive: bool
     ) -> 'cirq.CircuitOperation':
-        return self.with_params(resolver.param_dict, recursive)
+        resolved = self.with_params(resolver.param_dict, recursive)
+        return resolved.replace(repetitions=resolver.value_of(self.repetitions, recursive))
```

Comment:
An important clarification to make: if I'm reading this correctly, symbols in `repetitions` are only resolved using _external_ parameterization, i.e. they ignore `self.param_resolver`, but not the resolver in `cirq.resolve_parameters(self, resolver)`.

This seems like the right behavior to me, but it should be documented clearly (and maybe tested explicitly?).